scRNA-seq/05-dimension_reduction_scRNA-seq.Rmd

```diff
@@ -6,234 +6,396 @@ output:
+
+Then run the chunks and compare your output graphs.
+
+```{r run-UMAP-1, live = TRUE}
+# Try something low?
+UMAP_plot_wrapper(nn_param = 3)
 ```
 
-```{r Run t-SNE 4}
-# Run tSNE_plot_wrapper function for a fourth time, but use a different perplexity
-# argument
+```{r run-UMAP-2, live = TRUE}
+# Try something high?
+UMAP_plot_wrapper(nn_param = 100)
 ```
 
-```{r Run t-SNE 5}
-# Run tSNE_plot_wrapper function for a fifth time, but use a different perplexity
-# argument
+```{r run-UMAP-3, live = TRUE}
+# Try whatever you like!
+UMAP_plot_wrapper(nn_param = 5)
 ```
 
 #### Some 'big picture' thoughts to take from this experiment: 
 
-1) Different analyses (such as t-SNE) have various limitations for interpretability.
-   We found in our experiment that the coordinates of t-SNE output for any given 
-   cell might change drastically. 
-   This probably means that you shouldn't rely too heavily on the exact values of
-   t-SNE's output. 
-2) Different analyses also have their strengths.   
-   Using cell-type labeling, our experiment illustrated that t-SNE does 
-   appear to give some biologically relevant output information for this dataset. 
-3) Playing with parameters so you can fine-tune them is a good way to give you more
-   information about a particular analysis as well as the data itself. 
+1. Analyses (such as UMAP) have various limitations for interpretability. 
+The coordinates of UMAP output for any given cell can change dramatically depending on parameters, and even run to run with the same parameters. 
+This probably means that you shouldn't rely too heavily on the exact values of UMAP's output. 
+
+    - One particular limitation of UMAP and t-SNE is that while observed clusters have some meaning, the distance *between* clusters usually does not (nor does cluster density). 
+    The fact that two clusters are near each other should NOT be interpreted to mean that they are more related to each other than to more distant clusters. 
+    (There is some disagreement about whether UMAP distances have more meaning, but it is probably safer to assume they don't.)
+    
+2. Different analyses also have their strengths. 
+Using cell-type labeling, our experiment illustrated that UMAP does appear to give some biologically relevant output information for this dataset. 
+
+3. Playing with parameters so you can fine-tune them is a good way to give you more information about a particular analysis as well as the data itself. 
    
-In summary, if the results of an analysis can be completely changed by changing its
-parameters, you should be more cautious when it comes to the conclusions you
-draw from it as well as having good rationale for the parameters you choose. 
+In summary, if the results of an analysis can be completely changed by changing its parameters, you should be more cautious when it comes to the conclusions you draw from it as well as having good rationale for the parameters you choose. 
+
+### PCA comparison
```

Comment:
Agree with moving this plot up (provided that is, in fact, what you meant in Slack)